src/coreclr/vm/prestub.cpp

```diff
@@ -719,6 +719,15 @@ namespace
 
         COR_ILMETHOD_DECODER* pHeader = NULL;
         COR_ILMETHOD* ilHeader = pConfig->GetILHeader();
+
```

Comment:
What we do here: 
when pConfig tries to match the native code to global IL it will follow `native code`->`methoddef`->`methoddesc`->IL   chain. 
However, we have two methoddescs associated with the methoddef and the one that it the "primary" one (as in - stored in the def->desc mapping dictionary) is the thunk. It's signature matches metadata, but its IL is runtime-provided, thus there would be no global IL for it. 

In a case if we get NULL, we check if the original methoddesc was actually an async "implementation" helper that is not a wrapper/thunk. In such case we ask the methoddesc directly for its IL as that would be the IL we are looking for..